Core/ALife.Core/WorldObjects/Agents/Brains/NeuralNetworkBrains/NeuralNetworkBrainFactory.cs

```diff
@@ -0,0 +1,453 @@
+        {
+            Dictionary<string, double> neuronBiases = new Dictionary<string, double>();
+            foreach(Neuron n in oldLayer.Neurons)
+            {
+                neuronBiases.Add(n.Name, EvolveBetweenNegOneAndOne(n.Bias, oldBrain)); //Line diff
+            }
+            return neuronBiases;
+        }
+
+        private static double EvolveBetweenNegOneAndOne(double original, NeuralNetworkBrain oldBrain)
+        {
+            double val = original;
+            double shouldMod = Planet.World.NumberGen.NextDouble();
+            if(shouldMod < oldBrain.ModificationRate)
+            {
+                double rawMod = Planet.World.NumberGen.NextDouble();
+                double modification = ((rawMod * 2) - 1) * oldBrain.MutabilityRate;
+                val += modification;
+                val = ExtraMath.Clamp(val, -1, 1);
+            }
+            return val;
+        }
+
+        private static void CreateRandomDendritesForLayer(Layer layer, Layer parentLayer)
+        {
+            foreach(Neuron n in layer.Neurons)
+            {
+                foreach(Neuron parentNeuron in parentLayer.Neurons)
+                {
+                    Dendrite den = new Dendrite(parentNeuron);
+                    n.UpstreamDendrites.Add(den);
+                }
+            }
+        }
+
+        //TODO: DUPE OF CreateDendritesFromLayerTemplate
+        private static void CreateEvolvedDendritesFromLayerTemplate(List<Layer> newLayers, NeuralNetworkBrain templateBrain, int layerNum)
+        {
+            Layer templateLayer = templateBrain.Layers[layerNum]; //Line Diff
+
+            Layer parent = newLayers[layerNum - 1];
+            Layer current = newLayers[layerNum];
+
+            Dictionary<string, Neuron> parentNameToNeuron = PopulateNameToNeuron(parent.Neurons);
+            Dictionary<string, Neuron> templateNameToNeuron = PopulateNameToNeuron(templateLayer.Neurons);
+
+            foreach(Neuron n in current.Neurons)
+            {
+                if(!templateNameToNeuron.ContainsKey(n.Name))
+                {
+                    throw new Exception($"Neuron {n.Name} missing from template. Should have been caught earlier");
+                }
+                Neuron templateNeuron = templateNameToNeuron[n.Name];
+                foreach(Dendrite templateDendrite in templateNeuron.UpstreamDendrites)
+                {
+                    if(!parentNameToNeuron.ContainsKey(templateDendrite.TargetNeuronName))
+                    {
+                        throw new Exception($"Neuron {n.Name} missing from parent. Should have been caught earlier");
+                    }
+                    Dendrite newDendrite = new Dendrite(parentNameToNeuron[templateDendrite.TargetNeuronName]
+                                                        , EvolveBetweenNegOneAndOne(templateDendrite.Weight, templateBrain)); //Line Diff
```

Comment:
Let's add a TODO to clean up these methods to try and merge things to reduce code duplication.

Reply:
Added TODO